lms/static/coffee/src/instructor_dashboard/data_download.coffee

```diff
@@ -203,6 +208,52 @@ class ReportDownloads
     grid = new Slick.Grid($table_placeholder, report_downloads_data, columns, options)
     grid.autosizeColumns()
 
+    $delete_btns = @$section.find(".delete-report")
+    $delete_btns.click (e) =>
+      table_row = jQuery(e.target.parentElement.parentElement)
+      filename_cell = table_row.find(".course-forums-data")
+      @$file_to_delete = filename_cell.text()
+      if confirm gettext 'Are you sure you want to delete the file ' + @$file_to_delete + '? This cannot be undone.'
+        @$delete_element = @$section.find ".report-downloads-delete"
+        @$delete_endpoint = @$delete_element.data 'endpoint'
+        success_cb = =>
+          @remove_row_from_ui table_row
+          @display_file_delete_success()
+        failure_cb = =>
+          @display_file_delete_failure()
+        @delete_report(success_cb, failure_cb)
+
+  remove_row_from_ui: (row) ->
+    row_height = row.height()
+    rows_after = row.nextAll()
+    row.remove()
+    for sib_row in rows_after
+      $sib_row = jQuery(sib_row)
+      currX = $sib_row.offset().left
+      currY = $sib_row.offset().top
+      $sib_row.offset(top: currY - row_height, left: currX)
+
+  display_file_delete_success: ->
+    @$reports_request_response.text gettext('The file ' + @$file_to_delete + ' was successfully deleted.')
```

Comment:
I think we can do w/ out the instance state here. We can make `files_to_delete` a param to this function.

`success_cb` and `failure_cb` should create a closure on `files_to_delete` and pass it along to `display_file_delete_success`.